src/m365/spo/commands/sitedesign/sitedesign-get.ts

```diff
@@ -85,21 +85,19 @@ class SpoSiteDesignGetCommand extends SpoCommand {
       responseType: 'json'
     };
 
-    return request
-      .post<{ value: SiteDesign[] }>(requestOptions)
-      .then(response => {
-        const matchingSiteDesigns: SiteDesign[] = response.value.filter(x => x.Title === args.options.title);
+    const response: { value: SiteDesign[] } = await request.post<{ value: SiteDesign[] }>(requestOptions);
 
-        if (matchingSiteDesigns.length === 0) {
-          return Promise.reject(`The specified site design does not exist`);
-        }
+    const matchingSiteDesigns: SiteDesign[] = response.value.filter(x => x.Title === args.options.title);
 
-        if (matchingSiteDesigns.length > 1) {
-          return Promise.reject(`Multiple site designs with title ${args.options.title} found: ${matchingSiteDesigns.map(x => x.Id).join(', ')}`);
-        }
+    if (matchingSiteDesigns.length === 0) {
+      throw `The specified site design does not exist`;
+    }
 
-        return Promise.resolve(matchingSiteDesigns[0].Id);
-      });
+    if (matchingSiteDesigns.length > 1) {
+      throw `Multiple site designs with title ${args.options.title} found: ${matchingSiteDesigns.map(x => x.Id).join(', ')}`;
```

Comment:
let's wait with merging this for #5192 so that we may refactor this to use `handleMultipleResultsFound`

Reply:
the mentioned PR is finally merged. I think we may proceed with this 👍